maths/prime_number.rb

```diff
@@ -0,0 +1,31 @@
+# A ruby program to check a given number is prime or not
+# Mathematical explanation: A number which has only 2 factors i.e., 1 (one) and itself
+
+# Prime number check function
+def is_prime_number(number)
+  prime_flag = false
+  if number>0
+    if number == 1
+      prime_flag = true
+    else
+      prime_flag = (2..number/2).any? { |i| number % i == 0 }
```

Comment:
**[Optimization]**

```suggestion
      prime_flag = (2..number.sqrt).any? { |i| number % i == 0 }
```

I think `floor(√2)` can be used as the upper limit and it will reduce the amount of calculation.